### PR TITLE
Fix spelling mistakes across the whole repository

### PR DESCRIPTION
**Pull request by dcompoze**
**Update:** Pushed additional changes based on the review comments.

**This pull request fixes various spelling mistakes in this repository.**

Most of the changes are contained in the first **3** commits:

- `Fix spelling mistakes in comments and docs`

- `Fix spelling mistakes in test names`

- `Fix spelling mistakes in error messages, panic messages, logs and tracing`

Other source code spelling mistakes are separated into individual commits for easier reviewing:

- `Fix the spelling of 'authority'`

- `Fix the spelling of 'REASONABLE_HEADERS_IN_JUSTIFICATION_ANCESTRY'`

- `Fix the spelling of 'prev_enqueud_messages'`

- `Fix the spelling of 'endpoint'`

- `Fix the spelling of 'children'`

- `Fix the spelling of 'PenpalSiblingSovereignAccount'`

- `Fix the spelling of 'PenpalSudoAccount'`

- `Fix the spelling of 'insufficient'`

- `Fix the spelling of 'PalletXcmExtrinsicsBenchmark'`

- `Fix the spelling of 'subtracted'`

- `Fix the spelling of 'CandidatePendingAvailability'`

- `Fix the spelling of 'exclusive'`

- `Fix the spelling of 'until'`

- `Fix the spelling of 'discriminator'`

- `Fix the spelling of 'nonexistent'`

- `Fix the spelling of 'subsystem'`

- `Fix the spelling of 'indices'`

- `Fix the spelling of 'committed'`

- `Fix the spelling of 'topology'`

- `Fix the spelling of 'response'`

- `Fix the spelling of 'beneficiary'`

- `Fix the spelling of 'formatted'`

- `Fix the spelling of 'UNKNOWN_PROOF_REQUEST'`

- `Fix the spelling of 'succeeded'`

- `Fix the spelling of 'reopened'`

- `Fix the spelling of 'proposer'`

- `Fix the spelling of 'InstantiationNonce'`

- `Fix the spelling of 'depositor'`

- `Fix the spelling of 'expiration'`

- `Fix the spelling of 'phantom'`

- `Fix the spelling of 'AggregatedKeyValue'`

- `Fix the spelling of 'randomness'`

- `Fix the spelling of 'defendant'`

- `Fix the spelling of 'AquaticMammal'`

- `Fix the spelling of 'transactions'`

- `Fix the spelling of 'PassingTracingSubscriber'`

- `Fix the spelling of 'TxSignaturePayload'`

- `Fix the spelling of 'versioning'`

- `Fix the spelling of 'descendant'`

- `Fix the spelling of 'overridden'`

- `Fix the spelling of 'network'`

Let me know if this structure is adequate.

**Note:** The usage of the words `Merkle`, `Merkelize`, `Merklization`, `Merkelization`, `Merkleization`, is somewhat inconsistent but I left it as it is.

~~**Note:** In some places the term `Receival` is used to refer to message reception, IMO `Reception` is the correct word here, but I left it as it is.~~

~~**Note:** In some places the term `Overlayed` is used instead of the more acceptable version `Overlaid` but I also left it as it is.~~

~~**Note:** In some places the term `Applyable` is used instead of the correct version `Applicable` but I also left it as it is.~~

**Note:** Some usage of British vs American english e.g. `judgement` vs `judgment`, `initialise` vs `initialize`, `optimise` vs `optimize` etc. are both present in different places, but I suppose that's understandable given the number of contributors.

~~**Note:** There is a spelling mistake in `.github/CODEOWNERS` but it triggers errors in CI when I make changes to it, so I left it as it is.~~

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: cumulus/pallets/xcmp-queue/src/lib.rs [hunk 1]
```diff
@@ -462,7 +462,7 @@ impl<T: Config> Pallet<T> {
 		// Max message size refers to aggregates, or pages. Not to individual fragments.
 		let max_message_size = channel_info.max_message_size as usize;
 		let format_size = format.encoded_size();
-		// We check the encoded fragment length plus the format size agains the max message size
+		// We check the encoded fragment length plus the format size gains the max message size
```
Comment (joepetrowski) [inline cumulus/pallets/xcmp-queue/src/lib.rs:465]:
```suggestion
		// We check the encoded fragment length plus the format size against the max message size
```

File: cumulus/primitives/utility/src/lib.rs [hunk 2]
```diff
@@ -264,7 +263,7 @@ impl<
 }
 
 /// XCM fee depositor to which we implement the TakeRevenue trait
-/// It receives a Transact implemented argument, a 32 byte convertible acocuntId, and the fee
+/// It receives a Transact implemented argument, a 32 byte convertible accountId, and the fee
```
Comment (joepetrowski) [inline cumulus/primitives/utility/src/lib.rs:266]:
```suggestion
/// It receives a `Transact` implemented argument and a 32 byte convertible `AccountId`, and the fee
```

Comment (joepetrowski) [reply cumulus/primitives/utility/src/lib.rs:266]: Can't suggest on multiple lines here but the whole thing should be:
```rust
/// XCM fee depositor to which we implement the `TakeRevenue` trait.
/// It receives a `Transact` implemented argument and a 32 byte convertible `AccountId`, and the fee
/// receiver account's `FungiblesMutateAdapter` should be identical to that implemented by `WithdrawAsset`.
```

File: cumulus/primitives/utility/src/lib.rs [hunk 3]
```diff
@@ -763,7 +762,7 @@ mod test_trader {
 /// Implementation of `xcm_builder::EnsureDelivery` which helps to ensure delivery to the
 /// parent relay chain. Deposits existential deposit for origin (if needed).
 /// Deposits estimated fee to the origin account (if needed).
-/// Allows to trigger additional logic for specific `ParaId` (e.g. open HRMP channel) (if neeeded).
+/// Allows to trigger additional logic for specific `ParaId` (e.g. open HRMP channel) (if needed).
```
Comment (joepetrowski) [inline cumulus/primitives/utility/src/lib.rs:765]:
```suggestion
/// Allows triggering of additional logic for a specific `ParaId` (e.g. to open an HRMP channel) if needed.
```

File: polkadot/node/network/protocol/src/request_response/v1.rs [hunk 4]
```diff
@@ -183,7 +183,7 @@ impl IsRequest for AvailableDataFetchingRequest {
 pub struct StatementFetchingRequest {
 	/// Data needed to locate and identify the needed statement.
 	pub relay_parent: Hash,
-	/// Hash of candidate that was used create the `CommitedCandidateRecept`.
+	/// Hash of candidate that was used create the `CommittedCandidateRecept`.
```
Comment (joepetrowski) [inline polkadot/node/network/protocol/src/request_response/v1.rs:186]:
```suggestion
	/// Hash of candidate that was used create the `CommittedCandidateReceipt`.
```

File: polkadot/node/subsystem-bench/src/lib/approval/message_generator.rs [hunk 5]
```diff
@@ -560,12 +560,12 @@ struct TestSignInfo {
 	candidate_index: CandidateIndex,
 	/// The validator sending the assignments
 	validator_index: ValidatorIndex,
-	/// The assignments convering this candidate
+	/// The assignments covering this candidate
```
Comment (joepetrowski) [inline polkadot/node/subsystem-bench/src/lib/approval/message_generator.rs:563]: Maybe a follow up PR: Periods at the end of a sentence :). I don't know why people don't do it, but it's all over the place.

File: polkadot/zombienet_tests/smoke/0004-configure-broker.js [hunk 6]
```diff
@@ -22,7 +22,7 @@ async function run(nodeName, networkInfo, _jsArgs) {
       renewalBump: 10,
       contributionTimeout: 5,
     }),
-    // We need MOARE cores.
+    // We need MORE cores.
```
Comment (joepetrowski) [inline polkadot/zombienet_tests/smoke/0004-configure-broker.js:25]: I think this one was on purpose :)

File: substrate/frame/contracts/src/tests.rs [hunk 7]
```diff
@@ -5226,7 +5226,7 @@ fn cannot_set_code_indeterministic_code() {
 		.unwrap()
 		.account_id;
 
-		// We do not allow to set the code hash to a non determinstic wasm
+		// We do not allow to set the code hash to a non deterministic wasm
```
Comment (joepetrowski) [inline substrate/frame/contracts/src/tests.rs:5229]:
```suggestion
		// We do not allow to set the code hash to a non-deterministic wasm
```

File: substrate/frame/contracts/src/tests.rs [hunk 8]
```diff
@@ -5429,7 +5429,7 @@ fn locking_delegate_dependency_works() {
 			contract.storage_base_deposit() - ED
 		);
 
-		// Removing an unexisting dependency should fail.
+		// Removing an nonexisting dependency should fail.
```
Comment (joepetrowski) [inline substrate/frame/contracts/src/tests.rs:5432]:
```suggestion
		// Removing a nonexistent dependency should fail.
```

**Review by joepetrowski — APPROVED** [review]
🙌

**Comment by joepetrowski** [comment]
> Note: Some usage of British vs American english e.g. judgement vs judgment, initialise vs initialize, optimise vs optimize etc. are both present in different places, but I suppose that's understandable given the number of contributors.

Our general approach is to use "most common", but yeah some of these slip in. Here is a nice old reference for you: https://github.com/paritytech/substrate/pull/2138

I am a big fan of Ngram for these decisions, e.g. [initialize vs. initialise](https://books.google.com/ngrams/graph?content=initialize%2Cinitialise&year_start=1800&year_end=2019&corpus=en-2019&smoothing=3). Just pick the most common.

File: cumulus/primitives/utility/src/lib.rs (hunk 3 above)
Comment (dcompoze) [reply cumulus/primitives/utility/src/lib.rs:765]: Yeah, there's quite a lot of grammatical errors as well, though I mostly focused on spelling mistakes in this PR, maybe I'll make a follow up PR for this (though it's harder to detect grammatical errors with tools).

File: polkadot/node/subsystem-bench/src/lib/approval/message_generator.rs (hunk 5 above)
Comment (dcompoze) [reply polkadot/node/subsystem-bench/src/lib/approval/message_generator.rs:563]: Yeah, I wouldn't mind making another PR for periods at the end of sentences and also sentences in comments without capital letters at the beginning.

File: substrate/frame/contracts/src/tests.rs (hunk 7 above)
Comment (dcompoze) [reply substrate/frame/contracts/src/tests.rs]: Fixed

File: substrate/frame/contracts/src/tests.rs (hunk 8 above)
Comment (dcompoze) [reply substrate/frame/contracts/src/tests.rs]: Fixed

File: polkadot/node/network/protocol/src/request_response/v1.rs (hunk 4 above)
Comment (dcompoze) [reply polkadot/node/network/protocol/src/request_response/v1.rs]: Fixed

File: cumulus/primitives/utility/src/lib.rs (hunk 3 above)
Comment (dcompoze) [reply cumulus/primitives/utility/src/lib.rs]: Fixed

File: cumulus/primitives/utility/src/lib.rs (hunk 2 above)
Comment (dcompoze) [reply cumulus/primitives/utility/src/lib.rs]: Fixed

File: cumulus/pallets/xcmp-queue/src/lib.rs (hunk 1 above)
Comment (dcompoze) [reply cumulus/pallets/xcmp-queue/src/lib.rs]: Fixed

**Review by ggwpez — COMMENTED** [review]
This introduces breaking changes to lots of crates.  
I am not sure if we are doing our downstream teams a favour with this.  
It should *at least* mention all breaking changes in a [prdoc](https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/prdoc.md).

**Comment by dcompoze** [comment]
I can document all the crate API-level changes in a PRDoc, unless you want to revert those types of changes.

**Review by ggwpez — CHANGES_REQUESTED** [review]
>I can document all the crate API-level changes in a PRDoc, unless you want to revert those types of changes.

Yes i veto this on account of being unreviewable. There is no way to ensure that no storage-alias or similar was changed without going over everything manually.  
If it only updates doc comments then its okay IMHO.

**Comment by dcompoze** [comment]
So you want to revert all the code-level changes? @ggwpez 

What about code changes such as:

- Test and benchmark code (e.g. test function names)

- Code not directly related to polkadot/substrate e.g. bridges, cli tools, scripts

- Changes to tracing messages, error messages, panic messages and log messages

**Comment by ggwpez** [comment]
> I'm assuming the problematic changes are the ones that affect either the public API of crates or the blockchain runtime (i.e. different Wasm blob generated).

Yea. Im my personal opinion, we should only break APIs that are in use by downstream teams for good reasons. Internal stuff is fine.

**Comment by bkchr** [comment]
> Yea. Im my personal opinion, we should only break APIs that are in use by downstream teams for good reasons. Internal stuff is fine.

Maybe I have overseen it, but there wasn't that many breaking external changes? Mainly I remember the bridges stuff, which is probably only used by us and some of the metrics. What else?

**Comment by dcompoze** [comment]
I'm just going through the commits to check which changes affect the public API, so far found 1, will update in a sec.

**Comment by dcompoze** [comment]
OK, so I went through all the commits and found the changes that affect the public API in some way.

I documented them all in the PRDoc:

```yaml
# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json

title: Fix spelling mistakes in source code

doc:
  - audience: Node Operator
    description: |
      Some spelling mistakes in log output, error messages and tracing (prometheus/grafana) have been fixed.
  - audience: Runtime Dev
    description: |
      Public crate changes:
      - The public function `era_payout` in `polkadot/runtime/common` now accepts `total_stackable` instead of `total_stakable` as a parameter.
      - The phantom field in the public `ParentNumberAndHash` struct in `substrate/frame/merkle-mountain-range` is now called `_phantom` instead of `_phanthom`.
      - The public trait `RuntimeParameterStore` in `substrate/frame/support` had the associated type renamed from `AggregratedKeyValue` to `AggregatedKeyValue`.
      - The public trait `AggregratedKeyValue` in `substrate/frame/support` was similarly renamed to `AggregatedKeyValue`.
      - The public methods `test_versioning` and `test_versioning_register_only` of the `TestApi` trait in `substrate/primitives/runtime-interface/test-wasm` had the spelling of `versionning` changed to `versioning`.
      - The public functions `read_trie_first_descendant_value` and `read_child_trie_first_descendant_value` in `substrate/primitives/trie` had the spelling of `descedant` changed to `descendant`.
      - The public trait `Applyable` in `substrate/primitives/runtime` has been renamed to `Applicable`.
crates:
  - name: polkadot-runtime-common
  - name: pallet-mmr
  - name: frame-support
  - name: sp-runtime-interface-test-wasm
  - name: sp-trie
  - name: sp-runtime

```

Alternatively, if there is consensus here, I can just delete these changes.

**Review by ggwpez — APPROVED** [review]
The changes in `frame-support` are fine by me, since they are in experimental APIs.

The other stuff should probably be decided by @bkchr.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5645480

**Comment by dcompoze** [comment]
I have removed the `Applyable` vs `Applicable` and `Overlayed` vs `Overlaid` changes as they change quite a lot and also modify the public API.

I also rebased from `master` and fixed the CI clippy errors, but I'm not sure if I can Label the PR myself or if someone else has to do it.

**Comment by dcompoze** [comment]
Any idea why the TOML CI check wants me change `polkadot/node/subsystem-bench/Cargo.toml`:

From:
```toml
tokio = { version = "1.24.2", features = ["rt-multi-thread", "parking_lot"] }
```
To:
```toml
tokio = { version = "1.24.2", features = ["parking_lot", "rt-multi-thread"] }
```

Doesn't seem relevant to the changes in this PR.

**Comment by ggwpez** [comment]
> Also, are the `zombienet` and `westend` migration checks supposed to be failing?

Yea the westend one is known-to-fail and the zombienet tests are flaky.

**Comment by dcompoze** [comment]
Shouldn't the formatting change be part of the PR that introduced the changes? Namely https://github.com/paritytech/polkadot-sdk/pull/3698. I can format it here but that would be a commit that's unrelated to the changes in this PR.

**Comment by AndreiEres** [comment]
> Shouldn't the formatting change be part of the PR that introduced the changes? Namely #3698. I can format it here but that would be a commit that's unrelated to the changes in this PR.

Just create a new PR with the quickfix

**Comment by dcompoze** [comment]
Just did that ;) https://github.com/paritytech/polkadot-sdk/pull/3842

**Comment by dcompoze** [comment]
Re-based with the TOML formatting fix. 

What else needs to be done to merge this pull request? 

I'm assuming someone from `bridges-core` needs to also approve this?

**Review by acatangiu — APPROVED** [review]

**Comment by dcompoze** [comment]
Thanks everyone!

**Comment by AndreiEres** [comment]
Unfortunately, this PR brought back recently deleted files

**Comment by ggwpez** [comment]
Looks like `polkadot/node/subsystem-bench/src/lib/utils.rs` is the only file that it accidentally resurrected.

**Comment by dcompoze** [comment]
My bad, must've been from the last rebase.

**Comment by AndreiEres** [comment]
> Looks like `polkadot/node/subsystem-bench/src/lib/utils.rs` is the only file that it added

Great, then, I was afraid it could propagate to another ones. My file I already sorted out.

**Pull request merged**